go/chat/server.go

```diff
@@ -2078,21 +2078,104 @@ func (h *Server) AddTeamMemberAfterReset(ctx context.Context,
 }
 
 func (h *Server) SetConvRetentionLocal(ctx context.Context, arg chat1.SetConvRetentionLocalArg) (err error) {
-	defer h.Trace(ctx, func() error { return err }, "SetConvRetention(%v, %v)", arg.ConvID, arg.Policy.Summary())()
-	_, err = h.remoteClient().SetConvRetention(ctx, chat1.SetConvRetentionArg{
+	ctx = Context(ctx, h.G(), keybase1.TLFIdentifyBehavior_CHAT_GUI, nil, h.identNotifier)
+	defer h.Trace(ctx, func() error { return err }, "SetConvRetentionLocal(%v, %v)", arg.ConvID, arg.Policy.Summary())()
+	uid, err := utils.AssertLoggedInUID(ctx, h.G())
+	if err != nil {
+		return err
+	}
+	policy := arg.Policy
+	if _, err = h.remoteClient().SetConvRetention(ctx, chat1.SetConvRetentionArg{
 		ConvID: arg.ConvID,
-		Policy: arg.Policy,
+		Policy: policy,
+	}); err != nil {
+		return err
+	}
+
+	// Post a SYSTEM message to conversation about the change. If we're
+	// inheriting the team policy, fetch that for the message.
+	isInherit := false
+	typ, err := policy.Typ()
+	if err != nil {
+		return err
+	}
+	switch typ {
+	case chat1.RetentionPolicyType_INHERIT:
+		isInherit = true
+	}
+
+	p := chat1.Pagination{Num: 1}
+	ib, err := h.G().InboxSource.ReadUnverified(ctx, uid, types.InboxSourceDataSourceAll, &chat1.GetInboxQuery{
+		ConvID: &arg.ConvID,
+	}, &p)
+	if err != nil {
+		return err
+	}
+	if len(ib.ConvsUnverified) != 1 {
+		h.Debug(ctx, "no conversation found for retention policy SYSTEM message")
+		return nil
+	}
+	conv := ib.ConvsUnverified[0].Conv
+	if isInherit {
+		teamRetention := conv.TeamRetention
+		if teamRetention == nil {
+			policy = chat1.RetentionPolicy{}
+		} else {
+			policy = *teamRetention
+		}
+	}
+	membersType := conv.Metadata.MembersType
+	info, err := CreateNameInfoSource(ctx, h.G(), membersType).LookupName(
```

Comment:
Maybe this RPC should just take a name instead of ID

Reply:
we'd still have to do a lookup to send the convID to gregor so i don't think it buys us much